### PR TITLE
Feature/aqm restart

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

Tested the plot restart capability several times; works just fine on the second restart.  

Tested the fix for the PMAVE and OZMAX8 stats and it works fine as well.  

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

1)  Be sure that you copy over the /lfs/h2/emc/vpppg/emc.vpppg/evs/v1.0/prep/aqm data for the past few days.  
2)  Run the following:

qsub -v cyc=04 dev/drivers/scripts/jevs_aqm_grid2obs_stats.sh
qsub -v cyc=11 dev/drivers/scripts/jevs_aqm_grid2obs_stats.sh

This will test not only the name changes in the stat script files but also the stat generation of PMAVE and OZMAX8.   Those jobs should now run error-free.  I also want to see that the stmp working directory is in the right place this time.  

3) Run the plot job.  The first time, set the wallclock time to 20-30 minutes and let the job be killed (or just qdel the job).  Then reset the wallclock time to the full time and run again, to test the restart capability.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

Yes

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

8/25/23

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @PerryShafran-NOAA , I have cloned your fork and checked out branch `feature/aqm_restart`. Please also run sync fork on this branch.

**Comment by ShelleyMelchior-NOAA** [comment]
My initial attempt to run stats failed b/c of the mismatch in name for the aqm v6 grid2obs script. The file is named `exevs_aqm6_grid2obs_stats.sh` but the J-job is calling to run `exevs_aqmv6_grid2obs_stats.sh` Please either rename the script to include the `v` or edit the J-job to remove the `v`.

After  making the above correction (I renamed the exevs script), the stats processing completed nominally.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/aqm`
logfiles:
(cyc=04) `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o77712392`
(cyc=11) `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o77712398`

**Comment by AliciaBentley-NOAA** [comment]
I would suggest removing the version number from the script name entirely
since no version numbers are supposed to appear outside run.ver. Thanks!

___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by ShelleyMelchior-NOAA** [comment]
Testing of the plots restart processing is underway.
logfiles:
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o77712946` (qdel after ~6 mins, and confirmed png plots in the $DATA dir)
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o77713219`

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA @Ho-ChunHuang-NOAA @ShelleyMelchior-NOAA I don't know, but apparently there is different EVS processing for AQMv6 and AQMv7, which is why there are different files for each version.  Ho-Chun is away, but I do not know the difference in processing between v6 and v7.  I'll remove the v6 from the file right now, but we'll have to be aware of this if we need to make a change for v7.  

Perry

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA It makes sense that both version of scripts were kept around, especially since AQMv7 was supposed to be in before EVSv1 code delivery. Since it's not, we should use the AQMv6 versions and name them without and version numbers included. Also of note, I saw late last week that the /aqm headline .tar filename does not include a "lastXXdays" designation in it like the /global_det or /rtofs headline .tar filenames do. Could "lastXXdays" be added in this PR? Thanks.

**Comment by PerryShafran-NOAA** [comment]
OK, I will add those to this PR.  Ho-Chun did the headline scores and may not have been aware of the requirement.  Can you tell me if the rest of the filename is OK?  

Perry

**Comment by AliciaBentley-NOAA** [comment]
Hi Perry,

I will be attending the Ensemble Users Workshop today and will not be on
VPN in order to check the rest of the filename. If you could check the
filename conventions against the EVS documentation as part of this PR, that
would be great. Here is the document:

https://docs.google.com/document/d/1HOJ5hTCMz-RipkGLk8zd2dov3V5IbhtgGJocqbwGQX8/edit

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by PerryShafran-NOAA** [comment]
OK, thanks, I will do so.  I was just wondering if you remembered offhand if something else was missing beyond the "last31days" part of it.  Thanks!

Perry

**Comment by PerryShafran-NOAA** [comment]
Hi, @ShelleyMelchior-NOAA - I have made changes so that version numbers aren't being referred to in filenames.  They still exist in the repo in various places, but I wanted some tests to be done before I remove them completely.  They should be removed entirely within all codes; so I'm checking them to see if any actively used-files still have version numbers.  When AQMv7 becomes prod, then of course we'll need to adjust if need be, but we can't have both version numbered files in the repo.

Anyway, please test all three components of AQM in this set up.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Sure thing, @PerryShafran-NOAA . Any particular cycles? All cycles?

**Comment by PerryShafran-NOAA** [comment]
Cycles 4 and 11 would be good. Also I plan to delete all files with a version number in them, but after the tests are done.  There are some in the scripts directory and a few in parm.  

Perry

File: dev/drivers/scripts/aqm/prep/jevs_aqm_prep.sh [hunk 1]
```diff
@@ -55,21 +53,7 @@ cd $DATA
...
 export COMIN=/lfs/h2/emc/vpppg/noscrub/$USER/${NET}/${evs_ver}
-##
-## For aqmv7 NRT runs
-## export fcst_input_ver=v7
-## export COMINaqm=/lfs/h2/emc/ptmp/jianping.huang/emc.para/com/aqm/v7.0
-##
-## For AQMv6 restrospective output
-## export COMINaqm=/lfs/h2/emc/physics/noscrub/ho-chun.huang/verification/${MODELNAME}/${envir}
-##
-export COMOUT=$COMIN/${STEP}/${COMPONENT}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/aqm/prep/jevs_aqm_prep.sh:58]: Can you please bring COMOUT back into this driver? I suggest:
`export COMOUT=/lfs/h2/emc/vpppg/noscrub/${USER}/${NET}/${evs_ver}/${STEP}/${COMPONENT}`

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/prep/jevs_aqm_prep.sh:58]: Done.

**Comment by ShelleyMelchior-NOAA** [comment]
Prep seems to have completed well.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_prep_00.o78193727`
COMOUT= `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm/atmos.20230822/aqm`

**Comment by PerryShafran-NOAA** [comment]
Looks good.

Before you do stats, please be sure to bring in several days of prep directories from the emc.vpppg run.  

Perry

File: dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh [hunk 2]
```diff
@@ -40,56 +40,20 @@ export MODELNAME=aqm
...
-. ./PDY
-#
-## export KEEPDATA=NO
-#
-export VDATE=${PDYm3}
-#
+
 export COMOUT=$COMIN/${STEP}/${COMPONENT}
 export COMOUTsmall=${COMOUT}/${RUN}.${VDATE}/${MODELNAME}/${VERIF_CASE}
 export COMOUTfinal=${COMOUT}/${MODELNAME}.${VDATE}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh:59]: COMOUTsmall and COMOUTfinal are not being formed correctly b/c the declaration of VDATE has been removed. Either reinstate VDATE declaration in the driver or don't define COMOUTsmall and COMOUTfinal in the driver. For my testing I commented out COMOUTsmall and COMOUTfinal and allowed the process to adopt the default values defined in JEVS_AQM_STATS.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh:59]: Oh yes.  Let me fix that.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh:59]: Fixed.

**Comment by ShelleyMelchior-NOAA** [comment]
stats was re-run for cycles 04 and 11, using emc.vpppg prep files.
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o78196491` (cyc=04)
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o78197513` (cyc=011)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/aqm/aqm.20230822`

**Comment by PerryShafran-NOAA** [comment]
Hmm - COMOUT uses COMIN for its definition, which should be a no-no.  I'm not sure how that happened,, but that needs to be fixed.  COMOUT should not be emc.vpppg (which is why setting COMIN to emc.vpppg shouldn't be done for these tests).

I'll fix this.

Perry

**Comment by PerryShafran-NOAA** [comment]
This is fixed -please re-run.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Gah! Part of that might have been my oversight. I always fix that in the driver since I know I can't write into emc.vpppg. Sorry about that. I will run again.

**Comment by ShelleyMelchior-NOAA** [comment]
stats was re-run for cycles 04 and 11, using emc.vpppg prep files. Note the correct COMOUT this time.
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o78201088` (cyc=04)
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o78201427` (cyc=011)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/aqm.20230822`

**Comment by PerryShafran-NOAA** [comment]
This is good, this has no errors and the output is what we expect.

For a complete test, let's run plots; though we don't have to do it for restart as we tested that capability up above I think.  

Thanks!

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Correct! Restart was successfully demonstrated previously. I will simply run plots from start to finish w/ no disruption.

File: dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh:1]: Let's change COMOUT in this driver script, too.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh:1]: I'm changing COMOUT and also COMIN.  COMIN means data in - we shouldn't be creating a COMIN directory, so COMIN should never be ptmp.

**Comment by ShelleyMelchior-NOAA** [comment]
I ran plots through to completion. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230822`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o78205977`

**Comment by PerryShafran-NOAA** [comment]
The latest run of plots looks good.  

Perry

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
The code in this PR is approved for merge.

**Comment by PerryShafran-NOAA** [comment]
Before merge was done, I removed all the v6 and v7 specific files in scripts and in parm.  Do you think this needs another rerun of all the scripts, just in case?

**Comment by ShelleyMelchior-NOAA** [comment]
To test for the removal of v6 and v7 named files, I re-ran the stats processing for cyc=04, 11. I used the prep data from the EVS parallel.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o78762*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/`

**Comment by ShelleyMelchior-NOAA** [comment]
One more note, since I noticed a v6 and v7 change in `scripts/aqm/prep`, I decided to run prep for completeness. There were problems w/ where COMOUT is writing. Based on how PREP_SAVE_DIR is defined in `exevs_aqm_prep.sh`, I think `JEVS_AQM_PREP` needs to be modified to remove $RUN from the COMOUT definition.
Based on this ...
scripts/aqm/prep/exevs_aqm_prep.sh: `export PREP_SAVE_DIR=${COMOUT}/${RUN}.${VDATE}/${MODELNAME}`
remove $RUN from
`export COMOUT=${COMOUT:-$(compath.py ${envir}/com/${NET}/${evs_ver}/${STEP}/${COMPONENT}/${RUN})}`

I did my run by defining COMOUT in the driver to write to my $USER noscrub and left $RUN off the end. The output went to the desired location: `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm/atmos.20230827/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_prep_00.o78764175`

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

That is a bug you found.

export COMOUT=${COMOUT:-$(compath.py ${envir}/com/${NET}/${evs_ver}/${STEP}/${COMPONENT})}
is the correct statement in EVS/jobs/aqm/prep/JEVS_AQM_PREP

For code clean-up, please remove "## mkdir -p $ COMOUT.${VDATE}/${MODELNAME}" (line #100)
in EVS/scripts/aqm/prep/exevs_aqm_prep.sh

Ho-Chun

**Comment by PerryShafran-NOAA** [comment]
The prep output looks good; all files are in the output prep directory.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Perry confirmed the stats re-run looks good and confirms no v6 or v7 were identified in the log file output. This code is now ready for merge.

**Comment by Ho-ChunHuang-NOAA** [comment]
Before your merge of this PR, can Shelley tell me your current working directory path.  I want to double check the removal process including the "fcst_input_ver" dependence

ho-Chun

**Comment by ShelleyMelchior-NOAA** [comment]
@Ho-ChunHuang-NOAA  my HOMEevs=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/forks/PS/EVS`

**Comment by Ho-ChunHuang-NOAA** [comment]
@ShelleyMelchior-NOAA 

Other than minor code-clean-up.  I think this PR is good to go.   Thanks for letting me review it.

Ho-Chun

**Comment by ShelleyMelchior-NOAA** [comment]
I appreciate the extra eyeballs, @Ho-ChunHuang-NOAA ! The extra clean up can be included in a follow-on PR. There is also Lin Gan's aqm PR from his EIB review. I will get to that next, after merging this PR.

**Pull request merged**